sdk/core/azure-core/src/http/winhttp/win_http_transport.cpp

```diff
@@ -201,24 +201,27 @@ void GetErrorAndThrow(const std::string& exceptionMessage)
       exceptionMessage + " Error Code: " + std::to_string(error) + ".");
 }
 
-void WinHttpTransport::CreateSessionHandle(std::unique_ptr<_detail::HandleManager>& handleManager)
+void WinHttpTransport::CreateSessionHandle()
 {
-  // Use WinHttpOpen to obtain a session handle.
-  // The dwFlags is set to 0 - all WinHTTP functions are performed synchronously.
-  handleManager->m_sessionHandle = WinHttpOpen(
-      NULL, // Do not use a fallback user-agent string, and only rely on the header within the
-            // request itself.
-      WINHTTP_ACCESS_TYPE_NO_PROXY,
-      WINHTTP_NO_PROXY_NAME,
-      WINHTTP_NO_PROXY_BYPASS,
-      0);
-
-  if (!handleManager->m_sessionHandle)
+  if (!m_sessionHandle)
   {
-    // Errors include:
-    // ERROR_WINHTTP_INTERNAL_ERROR
-    // ERROR_NOT_ENOUGH_MEMORY
-    GetErrorAndThrow("Error while getting a session handle.");
+    // Use WinHttpOpen to obtain a session handle.
+    // The dwFlags is set to 0 - all WinHTTP functions are performed synchronously.
+    m_sessionHandle = WinHttpOpen(
```

Comment:
Not thread safe.

Reply:
I don't understand, how is thread safety changed at all with this change? Are you suggesting what's in master isn't thread safe either?